### ISSUE TITLE
Scroll issue when navigating between pages

### ISSUE THREAD

**Issue by gavinr** [sveltejs/kit#2760]
### Describe the bug

When scrolling down on a page then clicking a link, the next page is scrolled down when it should not be.

### Reproduction

1. Go to https://gavinr.github.io/sveltekit-scroll-issue/
2. Scroll down 75% on the page, click link ~400
3. Next page loads. 
    - Expected: page loads with scroll at top of page
    - Actual: page loads scrolled down to bottom of page.

### Logs

```shell
None
```

### System Info

```shell
System:
    OS: Have tested this on Windows 10 and Linux (Mint)
  Binaries:
    Node: 14.18.1 - /usr/bin/node
    npm: 8.1.1 - /usr/bin/npm
  Browsers:
    Have tested on chrome and Firefox (latest) - happens on both
  npmPackages:
    @sveltejs/adapter-static: ^1.0.0-next.21 => 1.0.0-next.21 
    @sveltejs/kit: next => 1.0.0-next.195 
    svelte: ^3.42.6 => 3.44.1
```

### Severity

serious

### Additional Information

1. See code at https://github.com/gavinr/sveltekit-scroll-issue
2. It looks like the issue started at  version `next.193`. For example, if you clone the above repository and run `npm install @sveltejs/kit@1.0.0-next.192` it will work correctly, but if you run `npm install @sveltejs/kit@1.0.0-next.193` it will be broken.
3. This is possibly the same issue as #2720 from @NewNik - I think that one was closed in error as you can see this clear replication case the scroll issues are still happening.

**Comment by f-elix** [comment]
I just tested this fix on version next.196 in a production project (in a dev environment) and I am still getting scrolling issues when navigating. 

It's very hard to pin down a particular condition because it happens inconsistently. Sometimes the same navigation that worked previously will stop working and vice-versa. I know that I'm not controlling the scroll in any of the components and the body has no margin. 

I noticed it happens more often when I scroll past the first half of the page before navigating to another page.

I reverted back to next.192 in the meantime, just wanted to let you know that the issue is still present.
